frontend_tests/node_tests/starred_messages.js

```diff
@@ -38,6 +40,48 @@ run_test("remove starred", (override) => {
     assert.equal(starred_messages.get_count(), 1);
 });
 
+run_test("get starred ids in topic", (override) => {
+    for (const id of [1, 2, 3, 4, 5]) {
+        starred_messages.starred_ids.add(id);
+    }
+
+    assert.deepEqual(starred_messages.get_starred_msg_ids_in_topic(undefined, "topic name"), []);
+    assert.deepEqual(starred_messages.get_starred_msg_ids_in_topic(3, undefined), []);
+
+    override(message_store, "get", (id) => {
```

Comment:
Can you add a brief comment explaining the design of this setup?  It's complex enough that I think it'd be useful for understanding the test, both for review now and folks working on the test in the future.

Reply:
(Also, would it be simpler to just insert messages 2-5, maybe with a few more fields added, into `message_store`, rather than mocking the `get` function?  Generally we prefer that sort of test where convenient).

If that doesn't feel easy, we may want to add a common test helper `create_mock_message()` that takes just id/type/recipient as parameters and fills out plausible values for all the other fields to make a real message object; I'm sure we'd find many other places to use such a helper.

Reply:
I added the `create_mock_message` helper.

> and fills out plausible values for all the other fields

I left this part out, the reasoning being that it would reduce the utility of this helper, and I think it would be better for test readability to just add all fields required in the test itself. I've written more about this in the commit message, but please correct me if I'm wrong! In this commit I also changed some existing instances of `messa_store.get()` mocks to use this new helper method. One more thing -- I added this helper in `static/message_store.js` itself, should it be somewhere else, or this is a fine place for it? (I added a comment in there saying that its for use in tests only, though)